packages/block-library/src/navigation/edit/utils.js

```diff
@@ -110,3 +120,10 @@ export function getNavigationChildBlockProps( innerBlocksColors ) {
 		},
 	};
 }
+
+export function getAllowedInnerBlocks() {
+	return applyFilters(
```

Comment:
The Navigation block has a list of `ALLOWED_BLOCKS`. If we're going to let developers insert their blocks as inner blocks here we should provide a filter to allow them to be added via the editor in case the user removes it during editing by accident.

Reply:
FWIW, we discussed early on during Block Hooks development that it might make sense to standardize an `allowedBlocks` property and make it e.g. a `block.json` field so it's also available to the Block Hooks mechanism (which can then decide against injecting a hooked block if it's not part of that list) 🤔